neo/core/spiketrainlist.py

```diff
@@ -222,10 +223,36 @@ def extend(self, iterable):
 
     @classmethod
     def from_spike_time_array(cls, spike_time_array, channel_id_array,
-                              all_channel_ids, units=None,
-                              t_start=None, t_stop=None, **annotations):
+                              all_channel_ids, t_stop, units=None,
+                              t_start=None, **annotations):
         """Create a SpikeTrainList object from an array of spike times
-        and an array of channel ids."""
+        and an array of channel ids.
+
+        *Required attributes/properties*:
+
+        :spike_time_array: (quantity array 1D, numpy array 1D, or list) The times of
+            all spikes.
+        :channel_id_array: (numpy array 1D, dtype int) The id of the channel (e.g. the
+            neuron) to which each spike belongs. This array should have the same length
+            as :attr:`spike_time_array`
+        :all_channel_ids`: (list, tuple, or numpy array 1D containing integers) All
+            channel ids. This is needed to represent channels in which there are no
+            spikes.
+        :units: (quantity units) Required if :attr:`spike_time_array` is not a
+                :class:`~quantities.Quantity`.
+        :t_stop: (quantity scalar, numpy scalar, or float) Time at which
+            spike recording ended. This will be converted to the
+            same units as :attr:`spike_time_array` or :attr:`units`.
+
+        *Recommended attributes/properties*:
+        :t_start: (quantity scalar, numpy scalar, or float) Time at which
+            spike recording began. This will be converted to the
+            same units as :attr:`spike_time_array` or :attr:`units`.
+            Default: 0.0 seconds.
```

Comment:
Would it make sense to make this default explicit and set is already in the method header?